packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md

```diff
@@ -0,0 +1,122 @@
+### 1: Start password recovery
+
+The user starts the password recovery flow with the following steps:
+
+1. Clicks the **Forgot Password?** link on the sign-in page.
+2. Enters their **Email or Username** in the dialog and then clicks **Next**.
+3. Chooses **Email** as the authenticator they want to use for password recovery and clicks **Submit**.
+
+Okta then tells the user to either click the link in the email or enter the code to continue and sends an email to their email address matching the Forgot Password template that was altered earlier.
+
+<div class="common-image-format">
+
+![Example of email sent to user](/img/advanced-use-cases/custom-pwd-recovery-custom-email.png "Password recovery email")
+
+</div>
+
+The email's **Reset Password** link includes the `otp` and `request.relayState` variables sent back as query parameters to the application. For instance, the URL in the email template,  `http://localhost:8080/magic-link/callback?otp=${oneTimePassword}&state=${request.relayState}`, might be rendered as `http://localhost:8080/magic-link/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32` in the email sent to the user.
+
+### 2: Handle the OTP and state parameters
+
+Create a callback handler method that takes the `otp` and `state` parameters in the query string and passes them as session parameters to a page that contains the Sign-In Widget. Before calling the callback handler method, retrieve the current `IDXClientContext` object from the browser session and check that `state` has a value.
+
+If `state` is `null` or the browser context object is `null` (because the user clicked the magic link from a different browser), the widget instructs advise the user to return to the original tab in the browser where they requested a password reset and enter the OTP to proceed.
+
+```java
+@GetMapping("/magic-link/callback")
+public ModelAndView handleMagicLinkCallback(
+   HttpServletRequest request,
+   @RequestParam(name = "state") String state,
+   @RequestParam(name = "otp") String otp,
+   HttpSession session) throws MalformedURLException
+{
+  if (session.getAttribute(IDX_CLIENT_CONTEXT) == null) {
+      try {
+         idxClientContext = idxAuthenticationWrapper.getClientContext();
+      }
+      catch (ProcessingException e) {
+         // return error details
+      }
+      session.setAttribute(IDX_CLIENT_CONTEXT, idxClientContext);
+  }
+
+  if (idxClientContext == null) {
+     // return error details
+  }
+
+  if (state == null) {
+      return new ModelAndView("redirect:" + oktaOAuth2Properties.getRedirectUri());
+  }
+
+```
+
+If both values are valid, pass them (along with all the values required by the widget) to the page hosting the widget in your application.
```

Comment:
"both values", which values?

> "If the values for ... are valid, pass them (and all the other required values) to the page in your app page that is hosting the widget.

It's also not clear that the parenthetical statement is necessary. It seems to be clear from context.

Reply:
Have deleted the whole text here and placed it at the top of the code block instead.